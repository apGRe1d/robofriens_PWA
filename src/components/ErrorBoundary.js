import React from 'react';

class ErrorBoundary extends React.Component {

  constructor() {
    super();
    this.state = {
      hasError: false
    }
  }

  // static getDerivedStateFromError(error) {
  //   return {hasError: true}
  // }

  componentDidCatch(error, errorInfo) {
    console.log({error});
    console.log({errorInfo});
    this.setState({hasError: true})
  }

  render() {
    if (this.state.hasError) {
      return <h1>Something went wrong..</h1>
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
