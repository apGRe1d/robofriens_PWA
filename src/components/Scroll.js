import React from 'react';

const Scroll = (props) => {
  let height = '800px';
  if (screen.height > 1080) {
    height = '1000px';
  }
  return (
    <div style={{ overflow: 'scroll', border: '5px solid black', height: height}}>
      {props.children}
    </div>
  );
};

export default Scroll;
