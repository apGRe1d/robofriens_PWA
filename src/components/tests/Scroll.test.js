import Scroll from '../Scroll';
import React from 'react';
import { shallow, mount } from 'enzyme';
import ErrorBoundary from '../ErrorBoundary';

it('Scroll test ', function () {
  const wrapper = shallow(<Scroll></Scroll>);
  expect(wrapper).toMatchSnapshot();

  Object.defineProperty(screen, 'height', {
    writable: true,
    configurable: true,
    value: 1100
  });
  expect(screen.height).toEqual(1100);

  const wrapper2 = shallow(<Scroll></Scroll>);
  expect(wrapper2).toMatchSnapshot();

});
