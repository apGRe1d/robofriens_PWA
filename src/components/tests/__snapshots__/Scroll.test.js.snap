// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll test  1`] = `
<div
  style={
    Object {
      "border": "5px solid black",
      "height": "800px",
      "overflow": "scroll",
    }
  }
/>
`;

exports[`Scroll test  2`] = `
<div
  style={
    Object {
      "border": "5px solid black",
      "height": "1000px",
      "overflow": "scroll",
    }
  }
/>
`;
